libs/@blockprotocol/graph/src/stdlib/interval.ts

```diff
@@ -0,0 +1,333 @@
+import {
+  NonNullTimeInterval,
+  TemporalBound,
+  Timestamp,
+  TimestampLimitedTemporalBound,
+} from "../types/temporal-versioning.js";
+import { boundIsAdjacentTo, compareBounds } from "./bound.js";
+
+/**
+ * Checks whether two given {@link NonNullTimeInterval}s are adjacent to one another, where adjacency is defined as
+ * being next to one another on the timeline, without any points between, *and where they are not overlapping*. Thus,
+ * if adjacent, the two intervals should span another given interval.
+ *
+ * @param lhs - The first interval of the comparison (order is unimportant)
+ * @param rhs - The second interval of the comparison
+ */
+export const intervalIsAdjacentToInterval = (
+  lhs: NonNullTimeInterval,
+  rhs: NonNullTimeInterval,
+): boolean => {
+  /*
+   Examples           |     1     |     2     |     3     |     4     |     5
+   ===================|===========|===========|===========|===========|===========
+   Interval A         | [---]     | [---)     | [---]     | [---)     | [-]
+   Interval B         |     (---] |     [---] |     [---] |     (---] |      [--]
+   -------------------|-----------|-----------|-----------|-----------|-----------
+   Contains Interval  |   true    |   true    |   false   |   false   |   false
+   */
+  return (
+    boundIsAdjacentTo(lhs.end, rhs.start) ||
+    boundIsAdjacentTo(lhs.start, rhs.end)
+  );
+};
+
+/**
+ * Returns whether or not the `rhs` {@link NonNullTimeInterval} is *completely contained* within the `lhs`
+ * {@link NonNullTimeInterval}.
+ *
+ * @param {NonNullTimeInterval} lhs - Checked if it contains the other
+ * @param {NonNullTimeInterval} rhs - Checked if it's contained _within_ the other
+ */
+export const intervalContainsInterval = (
+  lhs: NonNullTimeInterval,
+  rhs: NonNullTimeInterval,
```

Comment:
Because order does matter here, I would call them `outerInterval` and `innerInterval` or something else that makes it easier to track in the code – if we ever edit this it will make it easier to spot if they get mixed up.

Reply:
Happy to change that, just flagging that I'm never sure about naming variables like that when they're being used to check it. 

For example take a type-guard: 
```ts
isFish(pet: Pet): pet is Fish
```

I wouldn't want to call it:
 ```ts
isFish(fish: Pet): fish is Fish
```